packages/camera/camera_platform_interface/lib/src/types/camera_event.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+
+/// Event emitted from the platform implementation.
+class CameraEvent {
+  /// Creates an instance of [CameraEvent].
+  ///
+  /// The [eventType] argument is required.
+  ///
+  /// Depending on the [eventType], the [captureSize], [previewSize] and
+  /// [errorDescription] arguments can be null.
+  CameraEvent({
+    @required this.eventType,
+    this.captureSize,
+    this.previewSize,
+    this.errorDescription,
```

Comment:
(This is a problem of having a single class to hold all the event types: the amount of properties will tend to grow, unbounded, if new CameraEvent types are added later).

Reply:
Events were split to separate classes.